Algorithm.CSharp/PortfolioOptimizationNumericsAlgorithm.cs

```diff
@@ -0,0 +1,151 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using QuantConnect.Data;
+using MathNet.Numerics.LinearAlgebra;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This algorithm uses Math.NET Numerics library, specifically Linear Algebra object (Vector and Matrix) and operations, in order to solve a portfolio optimization problem.
+    /// </summary>
+    public class PortfolioOptimizationNumericsAlgorithm : QCAlgorithm
+    {
+        private const double _targetReturn = 0.1;
+        private const double _riskFreeRate = 0.01;
+        private double _lagrangeMultiplier;
+        private double _portfolioRisk;
+        private Dictionary<string, double> _mean;
+        private Dictionary<string, double> _stddev;
+        private Dictionary<string, double> _weights;
```

Comment:
These three values may make a nice 'SymbolData' class, instead of accessing parallel dictionaries. When possible, I try to avoid multiple parallel collections as they tend to hide the problem's natural objects.